nessus-agent/elx/configure.sls

```diff
@@ -0,0 +1,60 @@
+# This salt state installs the Nessus Agent using the provided
+# URL to the rpm source file.  Once installed, the agent will be
+# configured and linked to the Nessus server using the provided
+# parameters (Nessus 64 hexadecimal-digit key, Nessus server FQDN and
+# port, Nessus Agent Group Name).  The final step is to start tHe
+# Nessus Agent service.
+#################################################################
+{%- from tpldir ~ '/map.jinja' import nessus with context %}
+{%- set chkFile = '/etc/tenable_tag' %}
+{%- set staleDbs = salt.file.find('/opt/nessus_agent/var/nessus/', maxdepth='1', type='f', name='*.db') %}
+{%- set linkStr =
+        nessus.sbin_file ~
+        ' agent link --key=' ~ nessus.nessus_key ~
+        ' --host=' ~ nessus.nessus_server ~
+        ' --port=' ~ nessus.nessus_port ~
+        ' --groups=' ~ nessus.nessus_groups
+%}
+
+{%- if salt.file.file_exists(chkFile) %}
+Unlink Stale Agent-config:
+  cmd.run:
+    - name: '{{ nessus.sbin_file }} agent unlink'
+    - success_retcodes:
+      - 0
+      - 2
+    - unless: {{ linkStr }}
+
+Stop Nessus Agent:
+  service.dead:
+    - name: {{ nessus.package | lower }}
+    - require:
+      - cmd: Unlink Stale Agent-config
+
+  {%- for staleDb in staleDbs %}
+Nuke Stale {{ staleDb }} file:
+  file.absent:
+    - name: '{{ staleDb }}'
+    - require:
+      - Stop Nessus Agent
+    - require_in:
+      - Configure Nessus Agent
+  {%- endfor %}
+{%- endif %}
+
+Handle {{ chkFile }} file:
```

Comment:
Do we still need this state? If configuring the agent properly creates it, then we probably _don't_ want to arbitrarily remove it? Or do we need to shift this _back_ into the "if" clause above, since we already shifted the requisite from this state to `Configure Nessus Agent`...

Reply:
Absent the ability to do a meaningful connection-status query of the ACAS server, the presence of `/etc/tenable_tag` file is the only reliable indicator that one _needs_ to do the unlink action. But neither the ACAS operators were aware of nor did the product documentation yield any kind of "do you know me" query one could make.

Notionally, if anything's superfluous, it's probably the line-26 `unless:` (especially since there's reasons beyond "are there stale, baked-in files" that an attempt to link c/would fail).

Reply:
Ok, changed the `{%- if …` to look for the stale `.db` files instead of the `/etc/tenable_tag` file. The former more-reliably indicate a (potentially) stale configuration. The latter only contains a client-id string: if it's present but doesn't match, it causes one error; if it's present but does match, it causes a different error. Unfortunately, the exit-code looks to be the same, even if the error-string isn't: Saltstack _really_ wants a way to error-flag on exit-string for cases where the exit-code isn't as useful as it ought to be.